datafusion/expr/src/expr_fn.rs

```diff
@@ -180,6 +182,13 @@ pub fn approx_percentile_cont_with_weight(
     }
 }
 
+/// Create an EXISTS subquery expression
+pub fn exists(subquery: LogicalPlan) -> Expr {
```

Comment:
Maybe this function could take an `Arc<LogicalPlan>` so if someone already has one they can reuse it rather than making a copy to pass in here